data1/d1_read.c

```diff
@@ -459,13 +459,18 @@ data1_node *data1_mk_tag_data_zint (data1_handle dh, data1_node *at,
                                    NMEM nmem)
 {
     data1_node *node_data;
+    int sz = 0;
 
     node_data = data1_mk_tag_data (dh, at, tag, nmem);
     if (!node_data)
         return 0;
     node_data->u.data.what = DATA1I_num;
     node_data->u.data.data = node_data->lbuf;
-    sprintf (node_data->u.data.data, ZINT_FORMAT, num);
+    sz = snprintf (node_data->u.data.data, sizeof(node_data->u.data.data), ZINT_FORMAT, num);
+    if (sz >= sizeof(node_data->u.data.data)) {
+        node_data->u.data.data[sizeof(node_data->u.data.data)] = '\0';
+        yaz_log(YLOG_WARN, "buffer overflow for tag '%s', chars to print = %d, available = %d", tag, sz, sizeof(node_data->u.data.data));
+    }
```

Comment:
sizeof(u->u.data.data) is 8 on 64-bit machines. It's a pointer. The buffer, however, is node_data->lbuf; This has a size of `DATA1_LOCALDATA` = 12 in `data1.h`. It's not the record count that overflows .. It's most likely the a number of 12 digits that overflow (11 is OK with the trailing 0). I suspect the problem is the call to save "recordBytes" in zinfo.c. I take you indexed about 1 TB of raw record material?

In any case, great that you discovered the issue and hinted the problem. Thx for that.